python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -388,6 +399,75 @@ def terminate_scheduled_nodes(self):
         self.nodes_to_terminate = []
         self.update_worker_list()
 
+    def drain_nodes_via_gcs(self, provider_node_ids_to_drain: List[NodeID]):
+        """Send an RPC request to the GCS to drain (gracefully shutdown) the
+        nodes with the given node provider ids.
+        """
+        # The GCS expects Raylet ids in the request, rather than NodeProvider
+        # ids. To get the Raylet ids of the nodes to we're draining, we make
+        # the following translations of identifiers:
+        # node provider node id -> ip -> raylet id
```

Comment:
is this translation reliable?
I thought ip to id mapping is not one to one.

Reply:
Ehh, for situations where there are multiple node_ids at one ip, we require using a "use_node_id_as_ip" hack.
https://github.com/ray-project/ray/blob/893f57591df7cb7b1fb4fed9978604964123eead/python/ray/autoscaler/_private/monitor.py#L275-L281
So for the moment, we can pretend there's one id per ip.

Some refactoring along the lines of https://github.com/ray-project/ray/issues/19086 will help.

Reply:
maybe add a comment (or file and issue?)

Reply:
I think it's covered by https://github.com/ray-project/ray/issues/19086

Reply:
Will add a comment.